### PR TITLE
[flang][acc] Add support for lowering combined constructs

### PR DESCRIPTION
**Pull request by razvanlupusoru**
PR#80319 added support to record combined construct semantics via an attribute. Add lowering support for this.

### CODE REVIEW

File: flang/lib/Lower/OpenACC.cpp [hunk 1]
```diff
@@ -2015,6 +2017,11 @@ createLoopOp(Fortran::lower::AbstractConverter &converter,
   if (!collapseDeviceTypes.empty())
     loopOp.setCollapseDeviceTypeAttr(builder.getArrayAttr(collapseDeviceTypes));
 
+  if (combinedConstructs) {
```
Comment (clementval) [inline flang/lib/Lower/OpenACC.cpp:2020]: no braces for simple if

File: flang/lib/Lower/OpenACC.cpp [hunk 2]
```diff
@@ -2419,6 +2426,10 @@ createComputeOp(Fortran::lower::AbstractConverter &converter,
           mlir::ArrayAttr::get(builder.getContext(), firstPrivatizations));
   }
 
+  if (combinedConstructs) {
+    computeOp.setCombinedAttr(builder.getUnitAttr());
+  }
```
Comment (clementval) [inline flang/lib/Lower/OpenACC.cpp:2431]:
```suggestion
  if (combinedConstructs)
    computeOp.setCombinedAttr(builder.getUnitAttr());
```
no braces

**Review by clementval — APPROVED** [review]
LGTM. Just remove braces on the two simple if. Lowering is following llvm guidelines on this.

File: flang/lib/Lower/OpenACC.cpp (hunk 1 above)
Comment (razvanlupusoru) [reply flang/lib/Lower/OpenACC.cpp]: Done

File: flang/lib/Lower/OpenACC.cpp (hunk 2 above)
Comment (razvanlupusoru) [reply flang/lib/Lower/OpenACC.cpp]: Done

**Review by clementval — APPROVED** [review]

File: flang/lib/Lower/OpenACC.cpp (hunk 1 above)
Comment (clementval) [reply flang/lib/Lower/OpenACC.cpp]: Thanks!

**Review by vzakhari — APPROVED** [review]
Thank you, Razvan!

**Pull request merged**